src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
we could create `isErrorObject(result: any): result is typeof errorObject {...}` but that helper method may have performance implications, and I know that is a concern.